website/docs/r/storage_notification.html.markdown

```diff
@@ -0,0 +1,90 @@
+
+Creates a new notification configuration on a specified bucket, establishing a flow of event notifications from GCS to a Cloud Pub/Sub topic.
+ For more information see 
+[the official documentation](https://cloud.google.com/storage/docs/pubsub-notifications) 
+and 
+[API](https://cloud.google.com/storage/docs/json_api/v1/notifications).
+
+## Example Usage
+
+```hcl
+resource "google_storage_bucket" "bucket" {
+	name = "default_bucket"
+}
+		
+resource "google_pubsub_topic" "topic" {
+	name = "default_topic"
+}
+
+//In order to enable notifications,
+//a GCS service account unique to each project
+//must have the IAM permission "projects.topics.publish" to a Cloud Pub/Sub topic from this project
+//The only reference to this requirement can be found here:
+//https://cloud.google.com/storage/docs/gsutil/commands/notification
+//The GCS service account has the format of <project-id>@gs-project-accounts.iam.gserviceaccount.com
+//API for retrieving it https://cloud.google.com/storage/docs/json_api/v1/projects/serviceAccount/get
+
+resource "google_pubsub_topic_iam_binding" "binding" {
+	topic       = "${google_pubsub_topic.topic.name}"
+	role        = "roles/pubsub.publisher"
+		  
+	members     = ["serviceAccount:my-project-id@gs-project-accounts.iam.gserviceaccount.com"]
+}
+
+resource "google_storage_notification" "notification" {
+	bucket            = "${google_storage_bucket.bucket.name}"
+	payload_format    = "JSON_API_V1"
+	topic             = "${google_pubsub_topic.topic.id}"
+	event_types       = ["%s","%s"]
+	custom_attributes {
+		new-attribute = "new-attribute-value"
+	}
+	depends_on        = ["google_pubsub_topic_iam_binding.binding"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `bucket` - (Required) The name of the bucket.
+
+* `payload_format` - (Required) The desired content of the Payload. One of `"JSON_API_V1"` or `"NONE"`.
+
+* `topic` - (Required) The Cloud PubSub topic to which this subscription publishes.
+
+- - -
+
+* `custom_attributes` - (Optional)  A set of key/value attribute pairs to attach to each Cloud PubSub message published for this notification subscription
+
+* `event_types` - (Optional) List of event type filters for this notification config. If not specified, Cloud Storage will send notifications for all event types. The valid types are: `"OBJECT_FINALIZE"`, `"OBJECT_METADATA_UPDATE"`, `"OBJECT_DELETE"`, `"OBJECT_ARCHIVE"`
+
+* `object_name_prefix` - (Optional) Specifies a prefix path filter for this notification config. Cloud Storage will only send notifications for objects in this bucket whose names begin with the specified prefix.
+
+## Attributes Reference
+
+In addition to the arguments listed above, the following computed attributes are
+exported:
+
+* `self_link` - The URI of the created resource.
+
+## Import
+
+Storage notifications can be imported using the notification `id` in the format `<bucket_name>/notificationConfigs/<id>` e.g.
```

Comment:
I wonder whether it makes more sense to just have it be `<bucket_name>/<id>` that way people don't have to bother typing out "/notificationConfigs/" (unless there's somewhere where that string tends to be copy/pasted from). What do you think?

Also, do you think most people will generally know the id of the notification? If there can only be one notification per topic per bucket then you could use the topic name as something more easily identifiable. If not though then it probably has to be the id.

Reply:
I was debating on the way to  id them too.. The  reason I decided to go with the <bucket_name>/notificationConfigs/<id> is because that's how they are being returned in  CLI util:
```
gsutil  notifications list gs://test_bucket/
projects/_/buckets/test_bucket/notificationConfigs/89
        Cloud Pub/Sub topic: projects/sandbox/topics/test_topic
```
It seems to me that this would be one of the quickest and user friendly ways to get the list of notifications on the bucket, in case they need to be imported and the id can be just copy/pasted. What do you think?

Reply:
Yup, that's a great reason. Thanks for the explanation!